src/Features/LanguageServer/Protocol/Handler/Breakpoints/ValidateBreakableRangeHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.CodeAnalysis.Debugging;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    [ExportRoslynLanguagesLspRequestHandlerProvider(typeof(ValidateBreakableRangeHandler)), Shared]
+    [Method(LSP.VSInternalMethods.TextDocumentValidateBreakableRangeName)]
+    internal sealed class ValidateBreakableRangeHandler : AbstractStatelessRequestHandler<LSP.VSInternalValidateBreakableRangeParams, LSP.Range?>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ValidateBreakableRangeHandler()
+        {
+        }
+
+        public override bool MutatesSolutionState => false;
+        public override bool RequiresLSPSolution => true;
+
+        public override LSP.TextDocumentIdentifier? GetTextDocumentIdentifier(LSP.VSInternalValidateBreakableRangeParams request)
+            => request.TextDocument;
+
+        public override async Task<LSP.Range?> HandleRequestAsync(LSP.VSInternalValidateBreakableRangeParams request, RequestContext context, CancellationToken cancellationToken)
+        {
+            var document = context.Document;
+            Contract.ThrowIfNull(document);
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var span = ProtocolConversions.RangeToTextSpan(request.Range, text);
+            var breakpointService = document.Project.LanguageServices.GetRequiredService<IBreakpointResolutionService>();
```

Comment:
will this break for non-c#/vb @dibarbet shoudl we make this GetService and then return null if it's not available?

Reply:
I don't think its necessary - we should just not advertise the capability for other scenarios